lib/spack/spack/filesystem_view.py

```diff
@@ -98,6 +98,28 @@ def view_copy(src, dst, view, spec=None):
             )
 
 
+def view_func_parser(parsed_name):
+    # What method are we using for this view
+    if parsed_name in ("hardlink", "hard"):
+        link_fn = view_hardlink
+    elif parsed_name in ("copy", "relocate"):
+        link_fn = view_copy
+    else:
+        link_fn = view_symlink
+    return link_fn
```

Comment:
This should check for `'symlink'` and raise a `ValueError` for unknown values:

```suggestion
    if parsed_name in ("hardlink", "hard"):
        return view_hardlink
    elif parsed_name in ("copy", "relocate"):
        return view_copy
    elif parsed_name == "symlink":
        return view_symlink
    else:
        raise ValueError("invalid link type for view: '%s'" % parsed_name)
```